tests/test_ecs.py

```diff
@@ -0,0 +1,1185 @@
+    """
+    Stop an ECS task.
+
+    Additional keyword arguments are passed to `ECSClient.stop_task`.
+    """
+    task = await run_sync_in_worker_thread(describe_task, ecs_client, task_arn)
+    # Check that the task started successfully
+    assert task["lastStatus"] == "RUNNING", "Task should be RUNNING before stopping"
+    print("Stopping task...")
+    await run_sync_in_worker_thread(ecs_client.stop_task, task=task_arn, **kwargs)
+
+
+def describe_task_definition(ecs_client, task):
+    return ecs_client.describe_task_definition(
+        taskDefinition=task["taskDefinitionArn"]
+    )["taskDefinition"]
+
+
+async def run_then_stop_task(task: ECSTask, **kwargs) -> str:
+    """
+    Run an ECS Task then stop it.
+
+    Moto will not advance the state of tasks, so `ECSTask.run` would hang forever if
+    the run is created successfully and not stopped.
+
+    Additional keyword arguments are passed to `ECSClient.stop_task`
+    """
+    session = task.aws_credentials.get_boto3_session()
+
+    with anyio.fail_after(20):
+        async with anyio.create_task_group() as tg:
+            task_arn = await tg.start(task.run)
+            # Stop the task after it starts to prevent the test from running forever
+            tg.start_soon(partial(stop_task, session.client("ecs"), task_arn, **kwargs))
+
+    return task_arn
+
+
+@pytest.fixture(autouse=True)
+def patch_task_watch_poll_interval(monkeypatch):
+    # Patch the poll interval to be way shorter for speed during testing!
+    monkeypatch.setattr(ECSTask.__fields__["task_watch_poll_interval"], "default", 0.05)
+
+
+@pytest.fixture
+def ecs_mocks(aws_credentials):
+    with mock_ecs() as ecs:
+        with mock_ec2():
+            inject_moto_patches(
+                ecs,
+                {
+                    # Add a container when describing any task
+                    "describe_tasks": [patch_describe_tasks_add_prefect_container],
+                    # Fix moto internal resource requirement calculations
+                    "_calculate_task_resource_requirements": [
+                        patch_calculate_task_resource_requirements
+                    ],
+                },
+            )
+
+            session = aws_credentials.get_boto3_session()
+            create_test_ecs_cluster(session.client("ecs"), "default")
+
+            # NOTE: Even when using FARGATE, moto requires container instances to be
+            #       registered. This differs from AWS behavior.
+            add_ec2_instance_to_ecs_cluster(session, "default")
+
+            yield ecs
+
+
+@pytest.mark.usefixtures("ecs_mocks")
+@pytest.mark.parametrize("launch_type", ["EC2", "FARGATE", "FARGATE_SPOT"])
+async def test_launch_types(aws_credentials, launch_type: str):
+    task = ECSTask(
+        aws_credentials=aws_credentials,
+        auto_deregister_task_definition=False,
+        command=["prefect", "version"],
+        launch_type=launch_type,
+    )
+    print(task.preview())
```

Comment:
Should we remove these print statements from the tests, or are they particularly useful when debugging test failures?

Reply:
Yeah I think they're useful on failure.